### ISSUE TITLE
Access BrowserSync instance for plugin (bs-pretty-message)

### ISSUE THREAD

**Issue by jpscharf** [JeffreyWay/laravel-mix#1553]
- Laravel Mix Version: 2.1.11 (`npm info laravel-mix version`)
- Node Version 6.11.5 (`node -v`): 
- NPM Version 5.7.1 (`npm -v`): 
- OS: Mac OS X 10.11.6

### Description:
I am trying to integrate [Kikobeats/bs-pretty-message](https://github.com/Kikobeats/bs-pretty-message) using the following setup:

```javascript
mix.js('resources/assets/js/app.js', 'public/js')
  …
  .browserSync({
    proxy: 'theperfumery.test',
    plugins: ['bs-pretty-message'],
  })
  .then(function() {
    if (stats.hasErrors() || stats.hasWarnings()) {
      // This is optional but chances are you'll want it.
      return browsersync.sockets.emit('fullscreen:message', {
        title: 'Error:',
        body: convert.toHtml(stats.toString({ colors: true })),
        timeout: 100000
      });
    } else {
      return browsersync.sockets.emit('styles:update');
    }
  });
```

But it does not have access to the BrowserSync instance.
- Is the BrowserSync instance created with a name, e.g. `const browsersync = require('browser-sync').create('instanceName')`? If so, what is it?
- Is there a way to access the BrowserSync instance?
- Would it be possible to set the name of the instance? So I could use `browsersync.get('instanceName')`?
- Would it be possible to expose the BrowserSync when using `mix.extend()`?

- As side note, `webpackConfig` has stats `webpackConfig.stats`, but it does not include the stats methods, i.e. `stats.hasErrors() || stats.hasWarnings()`
---
## Steps to get working in basic webpack ##
1. Require `const BrowserSyncPlugin = require('browser-sync-webpack-plugin');`
2. When initializing BrowserSync, give it a name:
```javascript
new BrowserSyncPlugin(
  {
     host: 'localhost',
     server: { baseDir: './' },
     files: 'index.html',
     plugins: ['bs-pretty-message'],
  }, {
     name: 'WebPackStarter',
     reload: false,
  }
),
```
3. Create a wwebpack plugin:
```javascript
function() {
    this.plugin('done', stats => {
        let browserSync = require('browser-sync').get('WebPackStarter')

        let chunks = stats.toJson().assets

        let files = chunks.map(chunk => {
            let hashed = chunk.name
            let name = ''
            let parts = hashed.split('.')
            if (parts.length == 2) {
                name = hashed
            } else {
                name = [parts[0], parts[parts.length - 1]].join('.')
            }
            return {
                hash: hashed,
                name
            }
        })

        require('fs').writeFileSync(
            path.join(__dirname, 'dist/manifest.json'),
            JSON.stringify(files)
        )

        if (stats.hasErrors() || stats.hasWarnings()) {
            return browserSync.sockets.emit('fullscreen:message', {
                title: 'Webpack Error:',
                body: stripAnsi(stats.toString()),
                timeout: 100000
            });
        }

        browserSync.reload();
    })
}]
```

**Comment by TankTheFrank** [comment]
I second this, would be nice to have access to the browsersync instance. I am using laravel-mix with go (I know, heresy!) and since is already watching for ui changes I also had it watch for go file changes and trigger the recompilation. This was straight forward to do but reloading the page once the compilation is complete and the binary restarted was not as I need to call reload() on the browsersync instance and couldn't figure out how to get it. I tried to follow it in the source code but lost the trail somewhere in a plugin factory.

As a workaround I am touching a template file when the build is finished that does trigger the regular templates/assets watch and gets the reload started. It works but it feels dirty (pun intended - 'man touch' if you don't get it).
